JoinColony/colonyCDapp
# Fix: handle loading state for the Crypto to Fiat tab components

## Description

![loader](https://github.com/user-attachments/assets/4df08601-c4db-4344-9533-49dca11a8eb1)

## Secrets setup

> [!IMPORTANT]
> If you haven't done the secrets file set up yet, expand this section:

<details>
<summary><h4>Click to expand and view the secrets setup 👈 🤫 </h4></summary>

### Setting up your local environment

Create a `.env.local.secrets` file in the parent directory and populate it with the correct environment variables:
```console
echo -e "BSCSCAN_API_KEY=\nETHERSCAN_API_KEY=\nGOOGLE_TAG_MANAGER_ID=\nPINATA_API_KEY=\nPINATA_API_SECRET=\nCOINGECKO_API_KEY=\nCOINGECKO_API_URL=\nPOSTHOG_KEY=\nPOSTHOG_HOST=\nUSDC_LOCAL_ADDRESS=\nPERSONA_ENVIRONMENT_ID=" > .env.local.secrets
```
Once the `.env.local.secrets` file is created, open it and you should see the following content. You will need to provide values for `POSTHOG_KEY` and `POSTHOG_HOST` so please ask a fellow dev about this.
```js
BSCSCAN_API_KEY=
ETHERSCAN_API_KEY=
GOOGLE_TAG_MANAGER_ID=
PINATA_API_KEY=
PINATA_API_SECRET=
COINGECKO_API_KEY=
COINGECKO_API_URL=
POSTHOG_KEY=
POSTHOG_HOST=
USDC_LOCAL_ADDRESS=
```
Once you've provided values for `POSTHOG_KEY` and `POSTHOG_HOST`, inject them:
```console
npm run prepare
```

</details>

## Testing

> [!IMPORTANT]
> Open the `amplify/backend/function/bridgeXYZMutation/src/index.js` file and make sure you enter the right sandbox api key. Otherwise, you won't be able to do the KYC flow:
> ```js
> // Use the sandbox API key instead of 'xxx'
> let apiKey = 'xxx';
> ```
> Enable the Crypto to Fiat tab via our Feature Flag. I'm sure there's a better way to do this but open the `src/context/FeatureFlagsContext/FeatureFlagsContextProvider.tsx` file and do the following replacement:
> ```js
> // From
> [FeatureFlag.CRYPTO_TO_FIAT]: cryptoToFiatFeature
> // To
> [FeatureFlag.CRYPTO_TO_FIAT]: { isEnabled: true }
> ```

### Testing steps
> [!NOTE]
> The KYC request is too fast when you have no data so you won't see the loader. We're gonna have to complete a KYC flow to make the request heavier.

1. Visit your account settings page
2. Click the Crypto to fiat tab
3. Complete Step 1 Verification
4. Once you're approved, just reload the page
5. Check that you can see the loading states

## Diffs

**New stuff** ✨

* A new HoC component that renders a configurable Skeleton loader when you give it a truthy value for its `isLoading` prop

**Changes** 🛠️

* The page is now using the `loading` state from the `useCheckKycStatusMutation` GraphQL call to detect if the request is busy

Resolves #2747
